yjit/src/core.rs

```diff
@@ -336,12 +345,7 @@ struct Branch {
     end_addr: Option<CodePtr>, // exclusive
 
     // Branch target blocks and their contexts
-    targets: [Option<BlockId>; 2],
-    target_ctxs: [Context; 2],
-    blocks: [Option<BlockRef>; 2],
-
-    // Jump target addresses
-    dst_addrs: [Option<CodePtr>; 2],
+    targets: [Option<Box<BranchTarget>>; 2],
```

Comment:
Would it be possible to just use a vec for the targets here? That would get rid of the option and box? Just make it length 1 when there is 1 target?

Reply:
It's possible, but would take some more refactoring and 24 bytes of storage (ptr, capacity, size) as opposed to the current 16.

Reply:
Are you sure it would take more storage? Because right now, Option probably adds 8 bytes, and Box uses 8 bytes for the pointer, and you have two Option<Box>.

Might be simpler than dealing with Option<Box> in terms of code?

Reply:
Good call to check! Yes it takes 16 bytes currently:
```
        assert_eq!(0, mem::size_of::<[Option<Box<BranchTarget>>; 2]>());
```
```
---- core::tests::types stdout ----
thread 'core::tests::types' panicked at 'assertion failed: `(left == right)`
  left: `0`,
 right: `16`', src/core.rs:2310:9
```
It's because of it knows Boxes are always none-NULL, so NULL is used to represent `None`.